### DISCUSSION TITLE
Failed to search collection

### DISCUSSION THREAD

**Discussion by brunowilliang** [pocketbase/pocketbase#1198] [Q&A]
I try to search for name like this and I get two results in my console.log()

the first: [ClientResponseError 0: Something went wrong while processing your request.]

the second it returns me the array with the data...

![Screen Shot 2022-12-07 at 10 40 11](https://user-images.githubusercontent.com/41432581/206194174-f8112615-f3f1-4874-974f-c568dc6ef06c.png)

![Screen Shot 2022-12-07 at 10 38 24](https://user-images.githubusercontent.com/41432581/206193881-f9040548-f318-4b6c-bff3-6001c2d0572f.png)

can someone help me?

**Comment by ganigeorgiev** [comment]
This is most likely a result of the autocancellation, aka. you are executing multiple requests without awaiting the previous one to complete and the SDK automatically cancels the pending requests and resolves only the last one.

You can handle this by `catch`-ing the error and ignore it if `err.isAbort` is `true`, although in react-native the check may not be correct since I'm not sure if the `DomException` error thrown by `AbortController` is available there.

You can disable completely the autocancellation behavior with `pb.autoCancellation(false)`.

You can find more information for the autocancellation and how to tweak it in https://github.com/pocketbase/js-sdk#auto-cancellation.

**Reply by brunowilliang** [reply]
hi @ganigeorgiev, I disabled it with `pb.autoCancellation(false)` but now it doesn't toast anymore (alert)

**Reply by ganigeorgiev** [reply]
Yes, this should be expected if the initial error was caused by the autocancellation since at least based from the code in the screenshot the toast (alert) is fired only in the `catch` error block.

While on the surface disabling autocancellation may seems to have fixed the issue, please note that your code is "racy", aka. your call stack is doing something like:

```js
searchCompany()
searchCompany()
searchCompany()
```

But there is no guarantee when each `searchCompany()` call will complete and in what order since the calls are not awaited (the `await` inside the `searchCompany()` function works only inside its `async` function block). What you want is:

```js
await searchCompany()
await searchCompany()
await searchCompany()
```

My recommendation would be to leave autocancellation enabled (because it will prevent the above "racy" behavior), catch the error and ignore it if `err.isAbort == true` or more generally with `err.status == 0` (0 means that it is a generic client-side error/exception; anything greater is a server-side error response), aka:

```js
...
.catch((err) => {
    if (err.status > 0) { // server-side error
        Toast({ ... })
    }
})
...
```
